### PR TITLE
Feature/fix tutorial form

### PR DESCRIPTION
**Pull request by HarisShahidHameed**
Trello-link : https://trello.com/c/mwFCtmv5/60-fix-tutorial-form-bugs

### CODE REVIEW

File: src/containers/Dashboard/Forms/TutorialForm.js [hunk 1]
```diff
@@ -72,15 +74,23 @@ class TutorialForm extends Component {
...
-        name: "",
-        content: "",
-        courseId: "",
-        tags : [],
-      })
+      
       API.tutorialData(data, result => {
         if (result.status === "201") {
-          this.formRef.reset();
+          self.setState({
```
Comment (abdulhaqshah) [inline src/containers/Dashboard/Forms/TutorialForm.js]: create a function for that to reset the values not here

**Review by abdulhaqshah — CHANGES_REQUESTED** [review]

File: src/containers/Dashboard/Forms/TutorialForm.js (hunk 1 above)
Comment (HarisShahidHameed) [reply src/containers/Dashboard/Forms/TutorialForm.js]: Done

File: src/containers/Dashboard/Forms/TutorialForm.js [hunk 2]
```diff
@@ -175,22 +188,41 @@ class TutorialForm extends Component {
...
-                    >
-                      -
-                    </button>
+                  
+                  <div className="tagholder" key={index}>
+                    <div className="row">
+                      <div className="col-xs-8 col-sm-8">
+                        <div>
+                        <input
+                          className="form-field-1"
```
Comment (abdulhaqshah) [inline src/containers/Dashboard/Forms/TutorialForm.js]: naming convention use proper

File: src/containers/Dashboard/Forms/TutorialForm.js [hunk 3]
```diff
@@ -202,7 +234,7 @@ class TutorialForm extends Component {
                     Add Tag
                 </button>
                 </div>
-                <div>
+                <div className="control-form">
```
Comment (abdulhaqshah) [inline src/containers/Dashboard/Forms/TutorialForm.js]: same as above

File: src/containers/Dashboard/Forms/TutorialForm.scss [hunk 4]
```diff
@@ -52,15 +65,25 @@
...
     .tagholder input {
       margin: 0;
       margin-top: 10px;
+      width: auto;
     }
     .tagholder-1 button {
       margin: 0;
       margin-top: 10px;
     }
+    .control-form {
```
Comment (abdulhaqshah) [inline src/containers/Dashboard/Forms/TutorialForm.scss]: same as above

File: src/containers/Dashboard/Forms/TutorialForm.js [hunk 5]
```diff
@@ -72,15 +83,18 @@ class TutorialForm extends Component {
...
-        tags : [],
-      })
+      
       API.tutorialData(data, result => {
         if (result.status === "201") {
-          this.formRef.reset();
+          this.setPreviousStates();
+          self.formRef.reset();
+          self.validator.hideMessages();
+          addNotification(
```
Comment (abdulhaqshah) [inline src/containers/Dashboard/Forms/TutorialForm.js]: in one line

File: src/containers/Dashboard/Forms/TutorialForm.js [hunk 6]
```diff
@@ -61,8 +63,17 @@ class TutorialForm extends Component {
     tags[i] = event.target.value;
     this.setState({ tags });
   }
+  setPreviousStates() {
```
Comment (abdulhaqshah) [inline src/containers/Dashboard/Forms/TutorialForm.js]: resetState

**Review by abdulhaqshah — CHANGES_REQUESTED** [review]

File: src/containers/Dashboard/Forms/TutorialForm.js (hunk 6 above)
Comment (HarisShahidHameed) [reply src/containers/Dashboard/Forms/TutorialForm.js]: Done.

File: src/containers/Dashboard/Forms/TutorialForm.js (hunk 5 above)
Comment (HarisShahidHameed) [reply src/containers/Dashboard/Forms/TutorialForm.js]: Done.

File: src/containers/Dashboard/Forms/TutorialForm.scss (hunk 4 above)
Comment (HarisShahidHameed) [reply src/containers/Dashboard/Forms/TutorialForm.scss]: Done.

File: src/containers/Dashboard/Forms/TutorialForm.js (hunk 3 above)
Comment (HarisShahidHameed) [reply src/containers/Dashboard/Forms/TutorialForm.js]: Done.

File: src/containers/Dashboard/Forms/TutorialForm.js (hunk 2 above)
Comment (HarisShahidHameed) [reply src/containers/Dashboard/Forms/TutorialForm.js]: Done.

**Review by abdulhaqshah — APPROVED** [review]

**Pull request merged**